### PR TITLE
PR for Brooke

### PR DESCRIPTION
**Pull request by eshipunova**
I’m speaking with myself, number one, because I have a very good brain and I’ve said a lot of things... I know what I’m doing and I listen to a lot of people, I talk to a lot of people and at the appropriate time I’ll tell you who the people are. But I speak to a lot of people. I always told people, you know I'm a very smart guy. I got good marks. 
(citation needed)

### CODE REVIEW

**Comment by bshima** [comment]
This is one of the most "memorable" presidential quotes since "fool me once shame on you, fool me ... you can't get fooled again".

**Comment by bshima** [comment]
Runs example successfully.  Please write a few more tests for edge cases (when there is no input, when there is a combo of text/integers, situations where there you'd expect failure, etc.)

File: my_main.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
+# HMMMM what should i do
+# I think book-reading should be an easy one. Or a math computation of some kind.
```
Comment (bshima) [inline my_main.py:2]: Remove comments that don't pertain to code

File: my_main.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+# Inputs: number
+# Outputs: all possible ways of adding digits together, printed out
+
+def print_all(n):
```
Comment (bshima) [inline my_main.py:6]: Add doc string to describe function purpose, parameters, returns

File: my_main.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    s = str(n)
```
Comment (bshima) [inline my_main.py:7]: Maybe make variable names more descriptive or add comments to clarify

File: my_main.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    for d in s:
+        # I should make some tea. Black tea? To stay awake. I need to make good progress by tomorrow
```
Comment (bshima) [inline my_main.py:9]: Remove comments that don't pertain to the code

File: my_main.py [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+        # 4|371730:
+        # 4 = 4, 3 + 7 + 1 + 7 + 3 + 0 = 21 =>
+        # 4 = 4, 2 + 1 = 3 =>
+        # 4 + 3 = 7
+        # 43|71730: 4 + 3 = 7, 7 + 1 + 7 + 3 + 0 = 18 => 7 = 7, 1 + 8 = 9 => 7 + 9 = 16 => 1 + 6 = 7
+        # 437|1730: 4 + 3 + 7 = 14, 1 + 7 + 3 + 0 = 11 => 1 + 4 = 5, 1 + 1 = 2 => 5 + 2 = 7
+        # 4371|730: 4 + 3 + 7 + 1 = 15, 7 + 3 + 0 = 10 => 1 + 5 = 6, 1 + 0 = 1 => 6 + 1 = 7
+        # It's cool that it's always 7. The proof is actually in the fact that we work with a decimal system but this is a neat little demonstration that it does work
+
+        # It smells. Suspiciously so
```
Comment (bshima) [inline my_main.py:26]: Remove comments that don't pertain to the code

File: my_main.py [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+        # Oh no, the kettle!!
+        raise Exception
```
Comment (bshima) [inline my_main.py:28]: Add exception text?

File: my_main.py [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+
+def print_summm(n):
```
Comment (bshima) [inline my_main.py:30]: Make function name more descriptive (or correct typo summm-->sum if this was not intended)

File: my_main.py [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+    s = str(n)
+    t = sum([int(d) for d in s])
+    return t
+
+def get_TWO_numbers_from_ONE_number(ALL_number,split_poInt):
```
Comment (bshima) [inline my_main.py:35]: Could you make this function name more concise?

File: my_main.py [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+    return FIRST_number, SECOND_number
+
+def split_and_print(numberone,numbertwo):
+    print("{} = {}, {} = {}".format(" + ".join(map(str, str(numberone))), print_summm(numberone),
+                                       " + ".join(map(str, str(numbertwo))),
+                                       print_summm(numbertwo)))
+    numberone = print_summm(numberone)
+    numbertwo = print_summm(numbertwo)
+    if len(str(numberone)) > 1 or len(str(numbertwo)) > 1:
+        #print(len(str(numberone)), len(str(numbertwo)))
```
Comment (bshima) [inline my_main.py:48]: Appears that this line was being used to test that the function was working, please remove

File: my_main.py [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+        split_and_print(numberone, numbertwo)
+    return print_summm(numberone), print_summm(numbertwo)
+
+def do_it_all(n):
+    n = str(n)
+    for i in range(len(n)-1):   # for 4371730 it would be 0,1,2,3,4,5
+        a_number, another_number_second_part = get_TWO_numbers_from_ONE_number(n,i)
+        print("{}|{}:".format(a_number, another_number_second_part))
+        #print(split_and_print(a_number, another_number_second_part))
```
Comment (bshima) [inline my_main.py:57]: Appears that this line was being used to test that the function was working, please remove

**Comment by bshima** [comment]
Overall, code look looks functional. Please consider adding docstring after the definition of each function for maximum clarity!

**Review by bshima — CHANGES_REQUESTED** [review]

File: my_main.py (hunk 6 above)
Comment (eshipunova) [reply my_main.py:28]: Yup, good catch!

**Comment by eshipunova** [comment]
First of all, thank you for appreciating my PR message :P I had a good time doing research for that one

You found a good amount of problems and even caught an issue with that exception and with bad inputs! You checked for tests and error handling and were very much on point with your comments
Some things I would recommend for the next time:
* Use "Suggested" and "Required" as a lead in to your comment, allowing the reviewee decide on as many things as it is reasonable
* Review structure. And by structure I mean file, function, class, etc organization. In this particular PR there was a fully empty file and an unused function. It can also be useful to think if the code is correctly distributed between functions? does anything repeat?
* If there is readme, it should serve a purpose
* Finally, it would be nice to have a more concise, all-encompassing summary, helping the reviewee to see the next steps
